### ISSUE TITLE
Table `start_line` off by one?

### ISSUE THREAD

**Issue by SamWilsn** [kivikakk/comrak#230]
I was poking around in comrak (thanks for building this btw!) to track `start_line` for inline nodes, and ran into some weirdness with tables. I wanted to confirm what the expected behaviour is.

My test case:

```markdown
| This | is | a | header |
| ---- | -- | - | ------ |
| row  | 1  | b | c      |
```

This parses into a `Table { start_line: 2, .. }`. Should the `start_line` be `1`, or is `2` indeed correct?

Thanks!

**Comment by kivikakk** [comment]
Ooh, thanks for the report. I think it should be `1`! I'll fix that up.

**Comment by SamWilsn** [comment]
Oh, wow, thanks for fixing it so quickly! Really appreciate it :3

I hate to be a pedant, but how do you feel about the top row's `TableRow` and `TableCell` nodes having a `start_line` of `2` still?

**Comment by kivikakk** [comment]
Oh my lord, I really just didn't even check, did I? I'll compare the cmark-gfm behaviour there too, one sec. :)

**Comment by SamWilsn** [comment]
If it helps, here's the test case I'm using:

```rust
    #[test]
    fn inlines_inside_table() {
        let arena = Arena::new();
        let input = r#"| This | is | a | header |
| ---- | -- | - | ------ |
| row  | 1  | b | c      |
"#;

        let config = ComrakOptions {
            extension: ComrakExtensionOptions {
                table: true,
                ..Default::default()
            },
            ..Default::default()
        };
        let lines = parse_document(&arena, input, &config);

        let lines: Vec<_> = lines
            .descendants()
            .filter(|n| matches!(n.data.borrow().value, NodeValue::Text(_)))
            .map(|n| n.data.borrow().start_line)
            .collect();

        assert_eq!(lines, [1, 1, 1, 1, 3, 3, 3, 3]);
    }
```

**Comment by kivikakk** [comment]
Thanks for that. I'm going to fix this right and add an XML formatter while I'm there, to make test cases for these kinds of things easier. Will be a little bit.

**Comment by kivikakk** [comment]
The eventual fix will come at #232, but if you feel like opening a PR for your specific case, please feel free.
